lops/lop-domain-a72.dts

```diff
@@ -73,8 +121,50 @@
                           for node1 in node_list:
                               if re.search('xlnx,axi-bram-ctrl', node1['compatible'].value[0]):
                                   invalid_nodes.append(node1)
+                          for node1 in node_list:
+                              match = 0
+                              for handle in valid_phandles:
+                                  if handle == node1.phandle:
+                                      match += 1
+                              if match == 0:
+                                  invalid_nodes.append(node1)
                           for node1 in invalid_nodes:
                               tree.delete(node1)
+
+                          # Memory Domain handling
+                          if linux_mem_domain:
+                              try:
+                                  has_addr = linux_mem_domain['memory'].value
+                              except:
+                                  has_addr = []
+
+                              shared_mem = []
+                              try:
+                                  rsc_domain = linux_mem_domain['include'].value
+                                  for rsc in rsc_domain:
+                                      match = [node for node in domain_nodes if node.phandle == rsc]
+                                      if match:
+                                          try:
+                                              shared_mem = match[0]['memory'].value
+                                          except:
+                                              pass
+                              except:
+                                  pass
+
+                              if has_addr:
+                                  start_addr = has_addr[0]
+                                  mem_nodes = get_mem_nodes(tree)
+                                  for node1 in mem_nodes:
+                                      reg, size = scan_reg_size(node1, node1['reg'].value, 0)
+                                      if start_addr in range(reg, reg+size) and shared_mem:
+                                          linux_mem_domain['memory'].value.extend(shared_mem)
+                                          modify_val = update_mem_node(node1, linux_mem_domain['memory'].value)
+                                          node1['reg'].value = modify_val
+                                      elif start_addr in range(reg, reg+size):
+                                          modify_val = update_mem_node(node1, linux_mem_domain['memory'].value)
+                                          node1['reg'].value = modify_val
```

Comment:
We should move this block into lopper_lib as the start of a re-usable memory routine.  i.e. a routine to check the range and another to update it as appropriate.

Reply:
okay will update